File: app/src/main/java/com/study/android/morden/java/RoomViewModel.java
package com.study.android.morden.java;

import android.app.Application;
import android.os.AsyncTask;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.room.Room;

import java.util.List;

public class RoomViewModel extends AndroidViewModel {

    AppDatabase db;

    public RoomViewModel(@NonNull Application application) {
        super(application);
        db = Room.databaseBuilder(application, AppDatabase.class, "todo-db_java")
                .build();
    }

    public LiveData<List<Todo>> getAll() {
        return db.todoDao().getAll();
    }

    public void insert(Todo todo){
        new InsertAyncTask(db.todoDao()).execute(todo);
    }


    private static class InsertAyncTask extends AsyncTask<Todo, Void, Void> {
        private TodoDao mTodoDao;

        public InsertAyncTask(TodoDao mTodoDao) {
            this.mTodoDao = mTodoDao;
        }

        @Override
        protected Void doInBackground(Todo... todos) {
            mTodoDao.insert(todos[0]);
            return null;
        }
    }
}

File: app/src/main/java/com/study/android/morden/kotlin/RoomViewModel.kt
package com.study.android.morden.kotlin

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.viewModelScope
import androidx.room.Room
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch

class RoomViewModel(application: Application) : AndroidViewModel(application) {
    lateinit var todos: LiveData<List<Todo>>
    private val db: AppDatabase = Room.databaseBuilder(application, AppDatabase::class.java, "todo-db_kotlin").build()
    var newTodo: String? = null

    init {
        todos = getAll()
    }
    fun getAll(): LiveData<List<Todo>> {
        todos = db.todoDao().getAll()
        return todos
    }

    fun insert(todo: String) {
        viewModelScope.launch(Dispatchers.IO){
            db.todoDao().insert(Todo(todo))
        }
    }
}